ludwig/data/preprocessing.py

```diff
@@ -1089,6 +1157,9 @@ def build_dataset(
         backend
     )
 
+    for callback in callbacks or []:
+        callback.on_build_metadata_end(dataset_df)
+
```

Comment:
Can add callbacks for `on_build_data` below.

Reply:
Done.